src/test/java/picard/sam/SamErrorMetric/CollectSamErrorMetricsTest.java

```diff
@@ -128,15 +140,52 @@ private static void parseDirective0(final String directive, final String extensi
         }
     }
 
-    @DataProvider(name = "OneCovariateErrorMetricsDataProvider")
-    public Object[][] oneCovariateErrorMetricsDataProvider() {
-        final File simpleSamWithBaseErrors1 = new File(TEST_DIR, "simpleSamWithBaseErrors1.sam");
-        final File simpleSamWithBaseErrors2 = new File(TEST_DIR, "simpleSamWithBaseErrors2.sam");
-        final File simpleSingleStrandConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleSingleStrandConsensusSamWithBaseErrors.sam");
-        final File simpleDuplexConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleDuplexConsensusSamWithBaseErrors.sam");
-        final File chrMReadsWithClips = new File(TEST_DIR, "chrMReadsWithClips.sam");
-        final int priorQ = 30;
+    private final Map<File, File> errorMetrics = new HashMap<>();
+
+    private final File simpleSamWithBaseErrors1 = new File(TEST_DIR, "simpleSamWithBaseErrors1.sam");
+    private final File simpleSamWithBaseErrors2 = new File(TEST_DIR, "simpleSamWithBaseErrors2.sam");
+    private final File simpleSingleStrandConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleSingleStrandConsensusSamWithBaseErrors.sam");
+    private final File simpleDuplexConsensusSamWithBaseErrors = new File(TEST_DIR, "simpleDuplexConsensusSamWithBaseErrors.sam");
+    private final File chrMReadsWithClips = new File(TEST_DIR, "chrMReadsWithClips.sam");
+
+    @BeforeTest()
+    public void samMetricsProvider() {
+        final File[] files = new File[]{
+                simpleSamWithBaseErrors1,
+                simpleSamWithBaseErrors2,
+                simpleSingleStrandConsensusSamWithBaseErrors,
+                simpleDuplexConsensusSamWithBaseErrors,
+                chrMReadsWithClips};
+
+        OUTPUT_DATA_PATH = IOUtil.createTempDir("CollectSamErrorMetricsTest", null);
+
+        for (final File file : files) {
+            final File vcf = new File(TEST_DIR, "NIST.selected.vcf");
```

Comment:
Does this mean that `testdata/picard/sam/BamErrorMetrics/NIST.selected.vcf.idx` can be deleted?

Reply:
yes...but a slightly different path will be tested....